sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -229,8 +229,8 @@ def _receive(self, max_message_count=None, timeout=None):
         if len(batch) >= max_message_count:
             return [self._build_message(message) for message in batch]
 
-        # Dynamically issue link credit if max_message_count > 1 when the prefetch_count is the default value 1
-        if max_message_count and self._prefetch_count == 1 and max_message_count > 1:
+        # Dynamically issue link credit if the prefetch_count is the default value 1
+        if self._prefetch_count == 1:
```

Comment:
I think we could keep the old logic allowing `max_message_count = None` which has the special meaning that it's fully depending on the "prefetch" config on the receiver to decide on the receiving message count. For users they don't care/know the exact amount they want to receive but just want to consume the message as fast as possible. They could do:
```python
receiver = sbclient.get_quque_receiver(..., prefetch=1000)
with receiver:
    while more_message:
        messages = receiver.receive(max_message_count=None)
        # handle each message
        more_message = len(messages) > 0
```
Does this make sense to you?

Reply:
ah yes that sounds like a good thing to have there, perfect, I'll revert it